### DISCUSSION TITLE
Issue with storageManager  'remote'

### DISCUSSION THREAD

**Discussion by Double5peed** [GrapesJS/grapesjs#5709] [Q&A]
Hello All!

Looking to save/retrieve from MySQL so have created save.php and load.php and configured this in storageManager contentTypeJson: true. 

I've added debugging as you can see below I keep getting a storage error load:

Starting to load data...

news-letters/:64 Storage error (load):  SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON
    at JSON.parse (<anonymous>)
    at RemoteStorage.ts:81:56

news-letters/:56 Data loading process has ended.

Editor.ts:1051 SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON
    at JSON.parse (<anonymous>)
    at RemoteStorage.ts:81:56 {level: 'error'}

news-letters/:68 Direct fetch response (load.php): {"gjs-css": "div {color: red;}", "gjs-html": "<div>Hello World!</div>"}.

In developer tools under the network tab I've confirmed looking at load.php that the Response is:

{"gjs-css": "div {color: red;}", "gjs-html": "<div>Hello World!</div>"}.

Can anyone advise me what is wrong?

**Comment by artf** [comment]
Check the raw response, you're not returning a JSON but an HTML document.

**Reply by Double5peed** [reply]
Unfortunately, I am still experiencing the issue where the GrapesJS editor reports a SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON upon attempting to load data via the storageManager. This error suggests that the editor is expecting JSON but seems to be parsing an HTML document instead.

Based on the advice received here, I've conducted thorough testing to verify the nature of the response from our load.php. Here's a summary of the steps taken and the findings:

Direct Testing with tester.php: Implemented a PHP script (tester.php) that directly fetches the content from load.php using curl. This test consistently returns the correct JSON response, indicating that load.php is indeed serving JSON as expected.

(I also added a process to clean and validate the JSON content in load.php. This included trimming whitespace and validating the JSON structure to rule out issues related to formatting or encoding. Trimming Whitespace & JSON Validation).

Verbose cURL Output: The verbose output from tester.php confirms successful communication with the server, receiving a HTTP/1.1 200 OK response with Content-Type: application/json. The JSON returned matches the expected format, e.g., {"gjs-css": "div {color: red;}", "gjs-html": "<div>Hello World!</div>"}.

Direct Fetch within the Browser: Conducted tests directly in the browser using both incognito mode and normal mode to rule out caching issues. These tests also successfully retrieve JSON content from load.php.

GrapesJS Editor Load Error: Despite the successful direct tests, when GrapesJS attempts to load data, it encounters a JSON parsing error, suggesting it's receiving HTML content. This occurs even in scenarios where direct fetches from the same environment return valid JSON.

Given these detailed tests and consistent results showing load.php correctly returning JSON, I'm unsure why the GrapesJS editor experiences this JSON parsing error. The discrepancy between the direct fetch results and the GrapesJS load operation suggests a deeper issue possibly related to how the editor processes or interprets the response.

Can you provide any other suggestions on what might cause this behaviour in GrapesJS and how to resolve it. Could there be specific configurations or conditions within GrapesJS that I might have overlooked? Any advice or assistance from the community would be greatly appreciated.

Thank you for your time and support.

**Reply by artf** [reply]
I told you already, open the network tab from devtools and see the raw response from the load call.

**Reply by Double5peed** [reply]
![image](https://github.com/GrapesJS/grapesjs/assets/12670994/ac440f58-6340-49c5-93b0-694edb6e43fb)

**Reply by Double5peed** [reply]
{"gjs-css":"div {color: red;}","gjs-html":"<div>Hello World!<\/div>"}

**Reply by Double5peed** [reply]
Hello again,

Does the screen shot I provide help to show what may be wrong?
